.github/workflows/ci.yml

```diff
@@ -40,12 +40,61 @@ jobs:
             *.cache-from=type=gha,scope=build-${{ matrix.target }}
             *.cache-to=type=gha,scope=build-${{ matrix.target }}
       -
-        name: Upload build artifacts
-        uses: actions/upload-artifact@v2
+        name: Upload artifacts
+        uses: actions/upload-artifact@v3
         with:
           name: ${{ matrix.target }}
-          path: ${{ env.BUNDLES_OUTPUT }}/${{ matrix.target }}-daemon/*
+          path: ${{ env.BUNDLES_OUTPUT }}
+          if-no-files-found: error
+          retention-days: 7
+
+  cross:
```

Comment:
Should we remove the `cross` stage from Jenkins once this is merged?

remove `--build-arg CROSS=true` from https://github.com/moby/moby/blob/c6872980bb0b32d2f30f70c6761501a9cd31a94f/Jenkinsfile#L90

remove the `cross` stage: https://github.com/moby/moby/blob/c6872980bb0b32d2f30f70c6761501a9cd31a94f/Jenkinsfile#L167-L179

Reply:
I think we should remove the `CROSS` and `DOCKER_CROSSPLATFORMS` args in a follow-up and rely only on `--platform` like `docker/cli`, `buildx` so packaging will be easier and also we could remove the cross logic in the `.binary` script. But that needs some work to align the dependencies in the dockerfile.

Reply:
We need to make sure it can be run outside of the Dockerfile / container as well; it's possible there's people using the scripts to build/cross compile directly on the host.

(I guess xx can be used in that situation as well, but may need either extra instructions or a script to set it up)

Reply:
We do not need a non-Dockerfile "cross-all" type target if we have no use for it ourselves. The script for the single arch would run for any target arch inside Dockerfile(and cross-all is just `--platform` list). To run the same script locally it is user's responsibility that they have a cross toolchain for the target arch they are setting.

Reply:
> it's possible there's people using the scripts to build/cross compile directly on the host.

I don't know of distros that would use these scripts for cross-compilation, but I do know at least some distros definitely build their packages outside of containers/in their own environments.

Reply:
Let's continue that discussion when that change is actually made (this PR does not yet make that change)